### PR TITLE
feat(swap): add new quoting auth method [WEB-1494]

### PR DESCRIPTION
**Pull request by acdibble**

### CODE REVIEW

File: packages/swap/src/quoting/Quoter.ts [hunk 1]
```diff
@@ -97,11 +120,45 @@ export default class Quoter {
...
+    const quotedLegsMap = new Map<string, LegFormatter>();
+
+    for (const socket of await this.io.fetchSockets()) {
+      const [first, second] = request.legs;
+      const { quotedAssets, marketMaker } = socket.data;
+      const quotesFirstLeg = quotedAssets[first.getBaseAsset()];
+      const quotesSecondLeg = !second || quotedAssets[second.getBaseAsset()];
+      let message: MarketMakerQuoteRequest<LegJson>;
+
+      if (quotesFirstLeg && quotesSecondLeg) {
```
Comment (jerryafr) [inline packages/swap/src/quoting/Quoter.ts:140]: wouldn't `quotesSecondLeg` always either have value or be true?

**Review by GabrielBuragev — APPROVED** [review]

**Review by jerryafr — APPROVED** [review]

File: packages/swap/src/quoting/Quoter.ts [hunk 2]
```diff
@@ -97,11 +120,45 @@ export default class Quoter {
...
-    this.io.emit('quote_request', request);
+    let expectedResponses = 0;
+
+    const quotedLegsMap = new Map<string, LegFormatter>();
+
+    for (const socket of await this.io.fetchSockets()) {
+      const [first, second] = request.legs;
+      const { quotedAssets, marketMaker } = socket.data;
+      const quotesFirstLeg = quotedAssets[first.getBaseAsset()];
+      const quotesSecondLeg = !second || quotedAssets[second.getBaseAsset()];
```
Comment (niklasnatter) [inline packages/swap/src/quoting/Quoter.ts:137]: i dont understand the `!second` here but im sure there is a reason

**Review by niklasnatter — APPROVED** [review]

File: packages/swap/src/quoting/Quoter.ts (hunk 2 above)
Comment (acdibble) [reply packages/swap/src/quoting/Quoter.ts:137]: If there is no second leg

Comment (niklasnatter) [reply packages/swap/src/quoting/Quoter.ts:137]: but why do we need to set `quotesSecondLeg` to true then?

Comment (acdibble) [reply packages/swap/src/quoting/Quoter.ts:137]: So the first `if` statement below is used. Can probably simplify/make this more explicit

**Review by jerryafr — APPROVED** [review]

**Review by GabrielBuragev — APPROVED** [review]

**Review by niklasnatter — APPROVED** [review]
nice

**Pull request merged**